.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: ci
+
+on: [push, pull_request]
+
+jobs:
+  build_linux:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        build_config:
+          - { cc: gcc, cxx: g++, version: 10 }
+          - { cc: gcc, cxx: g++, version: 9 }
+          - { cc: gcc, cxx: g++, version: 8 }
+          - { cc: gcc, cxx: g++, version: 7 }
+          - { cc: clang, cxx: clang++, version: 11 }
+          - { cc: clang, cxx: clang++, version: 10 }
+          - { cc: clang, cxx: clang++, version: 9 }
+          - { cc: clang, cxx: clang++, version: 8 }
+          - { cc: clang, cxx: clang++, version: 7 }
+          - { cc: clang, cxx: clang++, version: 6 }
+    env:
+      CC: ${{ matrix.build_config.cc }}-${{ matrix.build_config.version }}
+      CXX: ${{ matrix.build_config.cxx }}-${{ matrix.build_config.version }}
+    name: "${{ matrix.build_config.cc }}-${{ matrix.build_config.version }}"
+    steps:
+      - uses: actions/checkout@master
+      - name: Setup Compiler
+        if: matrix.build_config.cc == 'clang'
+        run: |
+          sudo script/ci_install_clang.sh ${{ matrix.build_config.version }}
+          if [ "${{ matrix.build_config.version }}" -ge "7" ]; then echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV; fi
+      - name: Setup
+        run: |
+          sudo script/ci_setup_dependencies.sh
+          sudo script/ci_setup_locales.sh
+      - name: Build
+        run: script/ci_build.sh
```

Comment:
You are not running the tests here. Should have an additional step:
```yml
      - name: Run
        working-directory: build
        run: ctest -C Release -j 4 --output-on-failure
```
Same for the OSX job.

Reply:
_See comments above._